src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/ShieldUtilitySpell.java

```diff
@@ -41,24 +42,25 @@ public ShieldUtilitySpell(Properties properties, ShieldUtilitySpellConfig config
         this.recovCooldown = config.RECOV_TICKS.get();
     }
 
-    @SubscribeEvent
-    public static void triggerSpell(LivingAttackEvent event) {
-        if (event.getEntityLiving() instanceof Player player && !player.getLevel().isClientSide() && event.getSource() instanceof EntityDamageSource) {
+    @SubscribeEvent(priority = EventPriority.LOW)
+    public static void triggerSpell(LivingHurtEvent event) {
+        if (event.getEntityLiving() instanceof Player player && !player.getLevel().isClientSide() && event.getSource() != DamageSource.STARVE && !event.getSource().isBypassInvul()) {
             ItemStack spellStack = getHighestEquippedSpellOfType(ShieldUtilitySpell.class, player);
             if (spellStack != null) {
                 ShieldUtilitySpell spell = (ShieldUtilitySpell) spellStack.getItem();
-                int hitsRemaining = spell.totalHits;
-                if (spellStack.hasTag()) {
-                    hitsRemaining = spellStack.getTag().getInt("minebound.shield_utility.hits_remaining");
-                }
+                CompoundTag tag = spellStack.getOrCreateTagElement("minebound.shield_utility");
+                float hitsRemaining = tag.getFloat("hits_remaining");
+                if (!tag.contains("hits_remaining")) 
+                    hitsRemaining = getShieldCount(player);
+        
                 if (hitsRemaining > 0) {
                     player.getLevel().playSound(null, player.blockPosition(), SoundEvents.SHIELD_BLOCK, SoundSource.PLAYERS, 1f, 1f);
                     event.setCanceled(true);
                     reduceMana(spell.manaCost, player);
 
-                    CompoundTag tag = spellStack.getOrCreateTag();
-                    tag.putInt("minebound.shield_utility.hits_remaining", hitsRemaining - 1);
-                    tag.putInt("minebound.shield_utility.cooldown", spell.recovCooldown);
+                    tag.putFloat("hits_remaining", hitsRemaining - event.getAmount());
```

Comment:
Maybe we should include some logic that checks if `hits_remaining` will go into the negative, in which case the remaining damage is applied to the player.

Reply:
Fixed